demo-circuit/src/common/data_structures.rs

```diff
@@ -57,6 +58,24 @@ impl WithdrawalCertificateData {
         }
     }
 
+    pub fn hash(&self) -> Result<FieldElement, Error> {
```

Comment:
Unfortunately `FieldHasher::hash()` has also a `personalization: Option<&[H::Data]>` field that we cannot propagate because the hash in zendoo-cctp-cryptolib ignore it. Moreover if we refactor the code to use this trait we'll have a lot of place where we build should use `hash(None)` ... that's is just worse than it is now. 

So we cannot implement the trait correctly and the refactored code will be a mess.

Reply:
Np !